### DISCUSSION TITLE
How to add proper `cmake` targets support for a package and get rid of message : `this is heuristically generated, and may not be correct`

### DISCUSSION THREAD

**Discussion by cppdev123** [microsoft/vcpkg#26944] [Q&A] [answered]
After installing a local package from zip archive `vcpkg` shows how to use the package from `cmake` and prints this line in the output: ` # this is heuristically generated, and may not be correct` although I provide a `package-config.cmake` file and its content: 

```
@PACKAGE_INIT@

include("${CMAKE_CURRENT_LIST_DIR}/package-targets.cmake")
include("${CMAKE_CURRENT_LIST_DIR}/package-config-version.cmake")

check_required_components(package)
```
and it works but I want to get rid of this `vcpkg` message

**Comment by Neumann-A** [comment, marked as answer]
use the cmd line option `--no-print-usage` 

> but I want to get rid of this vcpkg message

otherwise if you want to completely get rid of it provide an empty usage file or provide a usage file with the correct information. Supplying a usage file will get rid of the additional text `# this is heuristically generated, and may not be correct`

**Reply by cppdev123** [reply]
Thanks. I just noticed this usage file in some packages

**Reply by craigscott-crascit** [reply]
For others coming across this discussion, in your `portfile.cmake`, you literally write or install a file called `usage` to the expected location. For example:

```cmake
file(WRITE "${CURRENT_PACKAGES_DIR}/share/${PORT}/usage"
    "The port's usage instructions go here"
)
```

**Reply by craigscott-crascit** [reply]
What is the vcpkg release tag or commit hash at which support for a `usage` file became available?

**Reply by Neumann-A** [reply]
https://github.com/microsoft/vcpkg-tool/commit/0085a9db39340aa8619ada62433089c7d5b1697a
